apps/server/src/modules/account/services/account-idm.service.ts

```diff
@@ -0,0 +1,113 @@
+		throw new NotImplementedException();
+	}
+
+	// eslint-disable-next-line @typescript-eslint/require-await
+	async searchByUsernamePartialMatch(
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		userName: string,
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		skip: number,
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		limit: number
+	): Promise<{ accounts: AccountDto[]; total: number }> {
+		throw new NotImplementedException();
+	}
+
+	// eslint-disable-next-line @typescript-eslint/require-await, @typescript-eslint/no-unused-vars
+	async searchByUsernameExactMatch(userName: string): Promise<{ accounts: AccountDto[]; total: number }> {
+		throw new NotImplementedException();
+	}
+
+	// eslint-disable-next-line @typescript-eslint/require-await, @typescript-eslint/no-unused-vars
+	async updateLastTriedFailedLogin(accountId: EntityId, lastTriedFailedLogin: Date): Promise<AccountDto> {
+		throw new NotImplementedException();
+	}
+
+	async save(accountDto: AccountSaveDto): Promise<AccountDto> {
+		let accountId: string;
+		const idmAccount: IAccountUpdate = {
+			username: accountDto.username,
+			attRefTechnicalId: accountDto.idmReferenceId,
+			attRefFunctionalIntId: accountDto.userId,
+			attRefFunctionalExtId: accountDto.systemId,
+		};
+		if (accountDto.id) {
+			const id = await this.getInternalId(accountDto.id);
+			accountId = await this.identityManager.updateAccount(id, idmAccount);
+			if (accountDto.password) {
+				await this.identityManager.updateAccountPassword(id, accountDto.password);
+			}
+		} else {
+			accountId = await this.identityManager.createAccount(idmAccount, accountDto.password);
+		}
+
+		const updatedAccount = await this.identityManager.findAccountById(accountId);
+		return AccountIdmToDtoMapper.mapToDto(updatedAccount);
+	}
+
+	async updateUsername(accountRefId: EntityId, username: string): Promise<AccountDto> {
+		const id = await this.getInternalId(accountRefId);
+		await this.identityManager.updateAccount(id, { username });
+		const updatedAccount = await this.identityManager.findAccountById(id);
+		return AccountIdmToDtoMapper.mapToDto(updatedAccount);
+	}
+
+	async updatePassword(accountRefId: EntityId, password: string): Promise<AccountDto> {
+		const id = await this.getInternalId(accountRefId);
+		await this.identityManager.updateAccountPassword(id, password);
+		const updatedAccount = await this.identityManager.findAccountById(id);
+		return AccountIdmToDtoMapper.mapToDto(updatedAccount);
+	}
+
+	async delete(accountRefId: EntityId): Promise<void> {
+		const id = await this.getInternalId(accountRefId);
+		await this.identityManager.deleteAccountById(id);
+	}
+
+	async deleteByUserId(userId: EntityId): Promise<void> {
+		const idmAccount = await this.identityManager.findAccountByFctIntId(userId);
+		await this.identityManager.deleteAccountById(idmAccount.id);
+	}
+
+	private async getInternalId(accountRefId: string) {
```

Comment:
I was confused. I thought "internal" means internal to the Schulcloud, but now I assume it means internal to the idm-service. So this retrieves an Idm-Account-ID for a Schulcloud-Account-ID, correct?
If I understood it right, I would like to change this to `getIdmAccountId(schulcloudAccountId: string)` or something like that.

Reply:
Yep, thats way clearer. Done.